File: Solutions/0102- Binary Tree Level Order Traversal/Solution.py
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if not root:
            return []
        
        answer = []
        queue = [root]

        while queue:
            temp = []
            n = len(queue)
            while n > 0:
                node = queue.pop(0)
                temp.append(node.val)

                if node.left:
                    queue.append(node.left)
                
                if node.right:
                    queue.append(node.right)
                
                n -= 1

            answer.append(temp)
        
        return answer

File: Solutions/0102- Binary Tree Level Order Traversal/Solution.java
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        if (root == null)
            return new ArrayList<>();

        Queue<TreeNode> queue = new LinkedList<>();
        List<List<Integer>> answer = new ArrayList<>();
        queue.add(root);

        while (!queue.isEmpty()) {
            List<Integer> temp = new ArrayList<>();
            int n = queue.size();
            while (n > 0) {
                var node = queue.remove();
                temp.add(node.val);
                if (node.left != null) {
                    queue.add(node.left);
                }

                if (node.right != null) {
                    queue.add(node.right);
                }
                n--;

            }
            answer.add(temp);

        }

        return answer;

    }
}